designer/server/src/main/scala/pl/touk/nussknacker/ui/process/newdeployment/DeploymentService.scala

```diff
@@ -0,0 +1,147 @@
+    dbioRunner: DBIOActionRunner,
+    clock: Clock
+)(implicit ec: ExecutionContext) {
+
+  def processCommand(
+      command: DeploymentCommand
+  ): Future[Either[RunDeploymentError, Future[Option[ExternalDeploymentId]]]] = {
+    command match {
+      case command: RunDeploymentCommand =>
+        runDeployment(command)
+    }
+  }
+
+  private def runDeployment(
+      command: RunDeploymentCommand
+  ): Future[Either[RunDeploymentError, Future[Option[ExternalDeploymentId]]]] = {
+    dbioRunner.runInTransactionE(
+      (for {
+        scenarioMetadata <- getScenarioMetadata(command)
+        _ <- EitherT.fromEither(
+          Either.cond(command.user.can(scenarioMetadata.processCategory, Permission.Deploy), (), NoPermissionError)
+        )
+        _         <- saveDeployment(command, scenarioMetadata)
+        runResult <- invokeLegacyRunDeploymentLogic(command, scenarioMetadata)
+      } yield runResult).value
+    )
+  }
+
+  private def getScenarioMetadata(command: RunDeploymentCommand) =
+    EitherT[DB, RunDeploymentError, ProcessEntityData](
+      scenarioMetadataRepository
+        .getScenarioMetadata(command.scenarioName)
+        .map(_.map(Right(_)).getOrElse(Left(ScenarioNotFoundError(command.scenarioName))))
+    )
+
+  private def saveDeployment(command: RunDeploymentCommand, scenarioMetadata: ProcessEntityData) =
+    EitherT(
+      deploymentRepository.saveDeployment(
+        DeploymentEntityData(command.id, scenarioMetadata.id, Timestamp.from(clock.instant()), command.user.id)
+      )
+    ).leftMap { e => ConflictingDeploymentIdError(e.id) }
+
+  private def invokeLegacyRunDeploymentLogic(
+      command: RunDeploymentCommand,
+      scenarioMetadata: ProcessEntityData
+  ): EitherT[DB, RunDeploymentError, Future[Option[ExternalDeploymentId]]] = {
+    EitherT(
+      toEffectAll(
+        DB.from(
+          // TODO: Currently it doesn't use our deploymentId. Instead, it uses action id
+          legacyDeploymentService.processCommand(
+            LegacyRunDeploymentCommand(
+              CommonCommandData(
+                ProcessIdWithName(scenarioMetadata.id, command.scenarioName),
+                command.comment,
+                command.user
+              ),
+              savepointPath = None,
+              nodesDeploymentData = command.nodesDeploymentData,
+            )
+          )
+        ).asTry
+          .map(
+            _.map[Either[RunDeploymentError, Future[Option[ExternalDeploymentId]]]](Right(_))
+              .recover { case CommentValidationError(msg) =>
+                Left(NewCommentValidationError(msg))
+              }
+              .get
+          )
+      )
+    )
+  }
+
+  def getDeploymentStatus(
+      id: DeploymentId
+  )(implicit loggedUser: LoggedUser): Future[Either[GetDeploymentStatusError, StatusName]] =
+    dbioRunner.runInTransactionE(
+      (for {
+        deploymentWithScenarioMetadata <- EitherT[DB, GetDeploymentStatusError, DeploymentWithScenarioMetadata](
```

Comment:
what about this?
```scala
deploymentWithScenarioMetadata <- EitherT(deploymentRepository.getDeploymentById(id))
```
Won't it work?

Reply:
It is not relevant after changes around `DBIOActionRunner`

Reply:
For some reasons it didn't work but now it started to work :)